File: app/write_bytes.c
// Copyright (c) 2022 Archaeologic, Inc., Brad Richardson
// This software was developed for the U.S. Nuclear Regulatory Commission (US NRC) under contract # 31310020D0006:
// "Technical Assistance in Support of NRC Nuclear Regulatory Research for Materials, Waste, and Reactor Programs"
#include <stdint.h>
#include <stdio.h>

void write_32bits(int32_t * bits) {
    fwrite(bits, sizeof(int32_t), 1, stdout);
}

void write_64bits(int64_t * bits) {
    fwrite(bits, sizeof(int64_t), 1, stdout);
}

File: app/write_bytes.f90
! Copyright (c) 2022 Archaeologic, Inc., Brad Richardson
! This software was developed for the U.S. Nuclear Regulatory Commission (US NRC) under contract # 31310020D0006:
! "Technical Assistance in Support of NRC Nuclear Regulatory Research for Materials, Waste, and Reactor Programs"
module write_bytes
    use, intrinsic :: iso_c_binding, only: c_int32_t, c_int64_t

    implicit none
    private
    public :: write_32bits, write_64bits

    interface
        subroutine write_32bits(bits) bind(C)
            import c_int32_t
            implicit none
            integer(c_int32_t), intent(in) :: bits
        end subroutine

        subroutine write_64bits(bits) bind(C)
            import c_int64_t
            implicit none
            integer(c_int64_t), intent(in) :: bits
        end subroutine
    end interface
end module